symfony/plugins/orangehrmAttendancePlugin/Api/AttendanceEditPunchInRecordOverlapAPI.php

```diff
@@ -0,0 +1,239 @@
+                RequestParams::PARAM_TYPE_QUERY,
+                CommonParams::PARAMETER_EMP_NUMBER,
+                $this->getAuthUser()->getEmpNumber()
+            );
+
+            $recordId = $this->getRequestParams()->getInt(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::PARAMETER_RECORD_ID
+            );
+
+            list($punchInUtc, $punchOutUtc) = $this->getUTCTimeByOffsetAndDateTime();
+            $isPunchInOverlap = !$this->getAttendanceService()
+                ->getAttendanceDao()
+                ->checkForPunchInOverLappingRecordsWhenEditing($punchInUtc, $employeeNumber, $recordId, $punchOutUtc);
+
+            return new EndpointResourceResult(
+                ArrayModel::class,
+                [
+                    self::PARAMETER_IS_PUNCH_IN_OVERLAP => $isPunchInOverlap,
+                ]
+            );
+        } catch (AttendanceServiceException $attendanceServiceException) {
+            throw $this->getBadRequestException($attendanceServiceException->getMessage());
+        }
+    }
+
+    /**
+     * @return array
+     */
+    protected function getUTCTimeByOffsetAndDateTime(): array
+    {
+        $punchInDate = $this->getRequestParams()->getString(
+            RequestParams::PARAM_TYPE_QUERY,
+            self::PARAMETER_PUNCH_IN_DATE,
+        );
+
+        $punchOutDate = $this->getRequestParams()->getStringOrNull(
+            RequestParams::PARAM_TYPE_QUERY,
+            self::PARAMETER_PUNCH_OUT_DATE,
+        );
+
+        $punchInTime = $this->getRequestParams()->getString(
+            RequestParams::PARAM_TYPE_QUERY,
+            self::PARAMETER_PUNCH_IN_TIME,
+        );
+
+        $punchOutTime = $this->getRequestParams()->getStringOrNull(
+            RequestParams::PARAM_TYPE_QUERY,
+            self::PARAMETER_PUNCH_OUT_TIME,
+        );
+
+        $punchInTimezoneOffset = $this->getRequestParams()->getString(
+            RequestParams::PARAM_TYPE_QUERY,
+            self::PARAMETER_PUNCH_IN_TIME_ZONE_OFFSET,
+        );
+
+        $punchOutTimezoneOffset = $this->getRequestParams()->getStringOrNull(
+            RequestParams::PARAM_TYPE_QUERY,
+            self::PARAMETER_PUNCH_OUT_TIME_ZONE_OFFSET,
+        );
+
+        $punchInDateTime = $punchInDate . ' ' . $punchInTime;
+
+        $punchInDateTime = new DateTime(
+            $punchInDateTime,
+            $this->getDateTimeHelper()->getTimezoneByTimezoneOffset($punchInTimezoneOffset)
+        );
+
+        if (!is_null($punchOutDate)) {
+            $punchOutDateTime = $punchOutDate . ' ' . $punchOutTime;
+            $punchOutDateTime = new DateTime(
+                $punchOutDateTime,
+                $this->getDateTimeHelper()->getTimezoneByTimezoneOffset($punchOutTimezoneOffset)
+            );
+        } else {
+            // if open punch in record
+            $punchOutDateTime = new DateTime(date("Y-m-d", 0) . ' ' . date("H:i:s", 0));
```

Comment:
```suggestion
            $punchOutDateTime = new DateTime('now', new DateTimeZone(DateTimeHelperService::TIMEZONE_UTC));
                $punchOutDateTime->setTimestamp(0);
```
Shall we go like this?